In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
%cd /content/drive/MyDrive/Colab Notebooks/여친_TravelMate

/content/drive/MyDrive/Colab Notebooks/여친_TravelMate


In [ ]:
# pip install nest_asyncio

In [ ]:
# pip install aiohttp

# 0. 라이브러리 호출 및 데이터 로드

In [ ]:
import pandas as pd
import numpy as np
# import requests
# import asyncio
# import aiohttp
# from tqdm.asyncio import tqdm
# import nest_asyncio
from multiprocessing import Pool

In [ ]:
df = pd.read_csv("Data/Final/final_03.csv")

In [ ]:
df

,Unnamed: 0,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,...,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM,ADDR,RES_CAFE,VISIT_COUNT,VISIT_AREA_MAIN_TYPE,AVG_TIME_MIN
0,0,2208190004,d_d000296,14,산굼부리,2022-08-19 00:00:00,2022-08-19 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1,60.0
1,83,2208190002,d_d000296,12,한라산국립공원 관음사탐방로,2022-08-19 00:00:00,2022-08-19 00:00:00,NaN,제주특별자치도 제주시 오라이동 산 107-20,126.538006,...,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 오라이동 산 107-20,2,17,1,420.0
2,98,2208180010,d_d000296,10,한림리조트,2022-08-18 00:00:00,2022-08-19 00:00:00,제주특별자치도 제주시 한림읍 일주서로 5083-5,제주특별자치도 제주시 한림읍 협재리 1193,126.249956,...,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 한림읍 일주서로 5083-5,2,19,24,240.0
3,117,2208200001,d_d000296,17,스누피가든,2022-08-20 00:00:00,2022-08-20 00:00:00,제주특별자치도 제주시 구좌읍 금백조로 916,제주특별자치도 제주시 구좌읍 송당리 2303-1,126.779668,...,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 구좌읍 금백조로 916,2,135,6,130.0
4,170,2208180007,d_d000296,7,협재해수욕장,2022-08-18 00:00:00,2022-08-18 00:00:00,제주특별자치도 제주시 한림읍 한림로 329-10,제주특별자치도 제주시 한림읍 협재리 2447,126.239158,...,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 한림읍 한림로 329-10,2,272,1,60.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
34715,1300205,2208210008,d_d000621,6,장동민의크라우드pc방,2022-08-21 00:00:00,2022-08-21 00:00:00,제주특별자치도 제주시 건주로4길 10-4,제주특별자치도 제주시 도련일동 1939-3,126.579895,...,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 건주로4길 10-4,2,1,3,150.0
34716,1300206,2208230010,d_d000621,28,에코티엘,2022-08-23 00:00:00,2022-08-24 00:00:00,제주특별자치도 제주시 신광로4길 24,제주특별자치도 제주시 연동 251-69,126.487583,...,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 24,2,9,24,400.0
34717,1300215,2208230009,d_d000621,27,앤유 피시방,2022-08-23 00:00:00,2022-08-23 00:00:00,NaN,제주특별자치도 제주시 연동,126.496886,...,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 연동,2,1,3,210.0
34718,1300216,2208230007,d_d000621,25,전주아줌마두루치기,2022-08-23 00:00:00,2022-08-23 00:00:00,제주특별자치도 제주시 신광로4길 25,제주특별자치도 제주시 연동 251-59,126.487416,...,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 25,0,1,11,30.0


# 데이터 전처리

## 0. 식당 카페 분류 변경

In [ ]:
df.rename(columns = {'상권업종소분류명' : 'RES_CAFE'}, inplace = True)

In [ ]:
# 'RES_CAFE' 열에서 '미분류' 값을 2로 변경
df['RES_CAFE'] = df['RES_CAFE'].replace('미분류', 2)

# 'RES_CAFE' 열에서 '미분류' 값을 2로 변경
df['RES_CAFE'] = df['RES_CAFE'].replace('카페', 1)

In [ ]:
# 'RES_CAFE' 열의 데이터 타입을 int로 변경
df['RES_CAFE'] = df['RES_CAFE'].astype(int)

In [ ]:
df['RES_CAFE'].value_counts()

,count
RES_CAFE,
2,20306
0,11114
1,3300


In [ ]:
# df.to_csv('Data/Final/final_labeled.csv', index = False)

## 1. 제주도 데이터로 한정

In [ ]:
# 'ROAD_NM_ADDR' 열에 '제주'가 포함된 행만 선택 (결측값 무시)
jeju_df = df[(df['ROAD_NM_ADDR'].str.contains('제주', na=False)) | (df['LOTNO_ADDR'].str.contains('제주', na=False))]

In [ ]:
jeju_df

,Unnamed: 0,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,...,TRAVEL_STATUS_ACCOMPANY,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM,ADDR,RES_CAFE,VISIT_COUNT,VISIT_AREA_MAIN_TYPE
0,0,2208190004,d_d000296,14,산굼부리,2022-08-19 00:00:00,2022-08-19 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1
1,83,2208190002,d_d000296,12,한라산국립공원 관음사탐방로,2022-08-19 00:00:00,2022-08-19 00:00:00,NaN,제주특별자치도 제주시 오라이동 산 107-20,126.538006,...,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 오라이동 산 107-20,2,17,1
2,98,2208180010,d_d000296,10,한림리조트,2022-08-18 00:00:00,2022-08-19 00:00:00,제주특별자치도 제주시 한림읍 일주서로 5083-5,제주특별자치도 제주시 한림읍 협재리 1193,126.249956,...,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 한림읍 일주서로 5083-5,2,19,24
3,117,2208200001,d_d000296,17,스누피가든,2022-08-20 00:00:00,2022-08-20 00:00:00,제주특별자치도 제주시 구좌읍 금백조로 916,제주특별자치도 제주시 구좌읍 송당리 2303-1,126.779668,...,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 구좌읍 금백조로 916,2,135,6
4,170,2208180007,d_d000296,7,협재해수욕장,2022-08-18 00:00:00,2022-08-18 00:00:00,제주특별자치도 제주시 한림읍 한림로 329-10,제주특별자치도 제주시 한림읍 협재리 2447,126.239158,...,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 한림읍 한림로 329-10,2,272,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
34715,1300205,2208210008,d_d000621,6,장동민의크라우드pc방,2022-08-21 00:00:00,2022-08-21 00:00:00,제주특별자치도 제주시 건주로4길 10-4,제주특별자치도 제주시 도련일동 1939-3,126.579895,...,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 건주로4길 10-4,2,1,3
34716,1300206,2208230010,d_d000621,28,에코티엘,2022-08-23 00:00:00,2022-08-24 00:00:00,제주특별자치도 제주시 신광로4길 24,제주특별자치도 제주시 연동 251-69,126.487583,...,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 24,2,9,24
34717,1300215,2208230009,d_d000621,27,앤유 피시방,2022-08-23 00:00:00,2022-08-23 00:00:00,NaN,제주특별자치도 제주시 연동,126.496886,...,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 연동,2,1,3
34718,1300216,2208230007,d_d000621,25,전주아줌마두루치기,2022-08-23 00:00:00,2022-08-23 00:00:00,제주특별자치도 제주시 신광로4길 25,제주특별자치도 제주시 연동 251-59,126.487416,...,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 25,0,1,11


In [ ]:
# '제주'가 포함된 주소를 찾는 함수 정의
def find_jeju_address(row):
    if '제주' not in row['ADDR']:
        if '제주' in row['LOTNO_ADDR']:
            return row['LOTNO_ADDR']
        elif '제주' in row['ROAD_NM_ADDR']:
            return row['ROAD_NM_ADDR']
    return row['ADDR']

# apply 함수를 사용하여 'ADDR' 열 업데이트
jeju_df['ADDR'] = jeju_df.apply(find_jeju_address, axis=1)

In [ ]:
jeju_df[~jeju_df['ADDR'].str.contains('제주')]

,Unnamed: 0,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,...,TRAVEL_STATUS_ACCOMPANY,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM,ADDR,RES_CAFE,VISIT_COUNT,VISIT_AREA_MAIN_TYPE


## 1. VISIT_AREA_TYPE_CD(방문지 유형)

In [ ]:
df

,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,Y_COORD,...,TRAVEL_STYL_7,TRAVEL_STYL_8,TRAVEL_STATUS_RESIDENCE,TRAVEL_STATUS_DESTINATION,TRAVEL_STATUS_ACCOMPANY,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM
0,2208190004,d_d000296,14,산굼부리,2022-08-19,2022-08-19,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,33.432574,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
1,2208190002,d_d000296,12,한라산국립공원 관음사탐방로,2022-08-19,2022-08-19,NaN,제주특별자치도 제주시 오라이동 산 107-20,126.538006,33.390838,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
2,2208180010,d_d000296,10,한림리조트,2022-08-18,2022-08-19,제주특별자치도 제주시 한림읍 일주서로 5083-5,제주특별자치도 제주시 한림읍 협재리 1193,126.249956,33.389418,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
3,2208200001,d_d000296,17,스누피가든,2022-08-20,2022-08-20,제주특별자치도 제주시 구좌읍 금백조로 916,제주특별자치도 제주시 구좌읍 송당리 2303-1,126.779668,33.443359,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
4,2208180007,d_d000296,7,협재해수욕장,2022-08-18,2022-08-18,제주특별자치도 제주시 한림읍 한림로 329-10,제주특별자치도 제주시 한림읍 협재리 2447,126.239158,33.393866,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
38983,2208230010,d_d000621,28,에코티엘,2022-08-23,2022-08-24,제주특별자치도 제주시 신광로4길 24,제주특별자치도 제주시 연동 251-69,126.487583,33.489267,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1
38984,2208220008,d_d000621,15,성산포항종합여객터미널,2022-08-22,2022-08-22,제주특별자치도 서귀포시 성산읍 성산등용로 112-7,제주특별자치도 서귀포시 성산읍,126.934605,33.472891,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1
38985,2208230009,d_d000621,27,앤유 피시방,2022-08-23,2022-08-23,NaN,제주특별자치도 제주시 연동,NaN,NaN,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1
38986,2208230007,d_d000621,25,전주아줌마두루치기,2022-08-23,2022-08-23,제주특별자치도 제주시 신광로4길 25,제주특별자치도 제주시 연동 251-59,126.487416,33.488756,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34720 entries, 0 to 34719
Data columns (total 69 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   VISIT_AREA_ID              34720 non-null  int64  
 1   TRAVEL_ID                  34720 non-null  object 
 2   VISIT_ORDER                34720 non-null  int64  
 3   VISIT_AREA_NM              34720 non-null  object 
 4   VISIT_START_YMD            34720 non-null  object 
 5   VISIT_END_YMD              34720 non-null  object 
 6   ROAD_NM_ADDR               29536 non-null  object 
 7   LOTNO_ADDR                 34250 non-null  object 
 8   X_COORD                    34720 non-null  float64
 9   Y_COORD                    34720 non-null  float64
 10  ROAD_NM_CD                 0 non-null      float64
 11  LOTNO_CD                   0 non-null      float64
 12  POI_ID                     13740 non-null  object 
 13  POI_NM                     13740 non-null  obj

In [ ]:
drop_idx = [9, 12, 21, 22, 23]

In [ ]:
# 데이터프레임 필터링 함수
def filter_rows(df_chunk, drop_idx):
    return df_chunk[~df_chunk['VISIT_AREA_TYPE_CD'].isin(drop_idx)]

# 데이터프레임을 청크로 나누는 함수 (개선된 버전)
def split_dataframe(df, num_chunks):
    chunk_size = len(df) // num_chunks
    remainder = len(df) % num_chunks
    df_chunks = []

    start = 0
    for i in range(num_chunks):
        end = start + chunk_size + (1 if i < remainder else 0)
        df_chunks.append(df.iloc[start:end])
        start = end

    return df_chunks

# 병렬 처리 함수
def parallel_process(df, drop_idx, num_chunks):
    # 데이터프레임을 청크로 분할
    df_chunks = split_dataframe(df, num_chunks)

    # 청크별 데이터 누락 여부 확인
    for i, chunk in enumerate(df_chunks):
        print(f"Chunk {i} size: {len(chunk)}")

    # Pool 객체 생성
    with Pool(processes=num_chunks) as pool:
        # 각 청크에 대해 filter_rows 함수를 병렬로 적용
        results = pool.starmap(filter_rows, [(chunk, drop_idx) for chunk in df_chunks])

    # 필터링된 결과를 하나의 데이터프레임으로 결합
    return pd.concat(results)

In [ ]:
# 멀티프로세싱을 통해 데이터프레임 처리
num_chunks = 4
filtered_df = parallel_process(df, drop_idx, num_chunks)

Chunk 0 size: 9747
Chunk 1 size: 9747
Chunk 2 size: 9747
Chunk 3 size: 9747


In [ ]:
filtered_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 34812 entries, 0 to 38987
Data columns (total 67 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   VISIT_AREA_ID              34812 non-null  int64  
 1   TRAVEL_ID                  34812 non-null  object 
 2   VISIT_ORDER                34812 non-null  int64  
 3   VISIT_AREA_NM              34812 non-null  object 
 4   VISIT_START_YMD            34812 non-null  object 
 5   VISIT_END_YMD              34812 non-null  object 
 6   ROAD_NM_ADDR               29622 non-null  object 
 7   LOTNO_ADDR                 34325 non-null  object 
 8   X_COORD                    34573 non-null  float64
 9   Y_COORD                    34573 non-null  float64
 10  ROAD_NM_CD                 42 non-null     float64
 11  LOTNO_CD                   595 non-null    object 
 12  POI_ID                     13782 non-null  object 
 13  POI_NM                     13782 non-null  object 


In [ ]:
filtered_df.reset_index(inplace = True)
filtered_df

,index,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,...,TRAVEL_STYL_7,TRAVEL_STYL_8,TRAVEL_STATUS_RESIDENCE,TRAVEL_STATUS_DESTINATION,TRAVEL_STATUS_ACCOMPANY,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM
0,0,2208190004,d_d000296,14,산굼부리,2022-08-19,2022-08-19,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
1,1,2208190002,d_d000296,12,한라산국립공원 관음사탐방로,2022-08-19,2022-08-19,NaN,제주특별자치도 제주시 오라이동 산 107-20,126.538006,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
2,2,2208180010,d_d000296,10,한림리조트,2022-08-18,2022-08-19,제주특별자치도 제주시 한림읍 일주서로 5083-5,제주특별자치도 제주시 한림읍 협재리 1193,126.249956,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
3,3,2208200001,d_d000296,17,스누피가든,2022-08-20,2022-08-20,제주특별자치도 제주시 구좌읍 금백조로 916,제주특별자치도 제주시 구좌읍 송당리 2303-1,126.779668,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
4,4,2208180007,d_d000296,7,협재해수욕장,2022-08-18,2022-08-18,제주특별자치도 제주시 한림읍 한림로 329-10,제주특별자치도 제주시 한림읍 협재리 2447,126.239158,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
34807,38982,2208210008,d_d000621,6,장동민의크라우드pc방,2022-08-21,2022-08-21,제주특별자치도 제주시 건주로4길 10-4,제주특별자치도 제주시 도련일동 1939-3,126.579895,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1
34808,38983,2208230010,d_d000621,28,에코티엘,2022-08-23,2022-08-24,제주특별자치도 제주시 신광로4길 24,제주특별자치도 제주시 연동 251-69,126.487583,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1
34809,38985,2208230009,d_d000621,27,앤유 피시방,2022-08-23,2022-08-23,NaN,제주특별자치도 제주시 연동,NaN,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1
34810,38986,2208230007,d_d000621,25,전주아줌마두루치기,2022-08-23,2022-08-23,제주특별자치도 제주시 신광로4길 25,제주특별자치도 제주시 연동 251-59,126.487416,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1


In [ ]:
filtered_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34812 entries, 0 to 34811
Data columns (total 68 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   index                      34812 non-null  int64  
 1   VISIT_AREA_ID              34812 non-null  int64  
 2   TRAVEL_ID                  34812 non-null  object 
 3   VISIT_ORDER                34812 non-null  int64  
 4   VISIT_AREA_NM              34812 non-null  object 
 5   VISIT_START_YMD            34812 non-null  object 
 6   VISIT_END_YMD              34812 non-null  object 
 7   ROAD_NM_ADDR               29622 non-null  object 
 8   LOTNO_ADDR                 34325 non-null  object 
 9   X_COORD                    34573 non-null  float64
 10  Y_COORD                    34573 non-null  float64
 11  ROAD_NM_CD                 42 non-null     float64
 12  LOTNO_CD                   595 non-null    object 
 13  POI_ID                     13782 non-null  obj

In [ ]:
filtered_df['VISIT_AREA_TYPE_CD'].value_counts()

,count
VISIT_AREA_TYPE_CD,
11,12984
24,6529
1,5999
10,2103
4,1905
13,1133
7,1098
3,920
2,818


## 2. 만족도 결측치 삭제

In [ ]:
df_1 = filtered_df

In [ ]:
df_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34722 entries, 0 to 34721
Data columns (total 68 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   index                      34722 non-null  int64  
 1   VISIT_AREA_ID              34722 non-null  int64  
 2   TRAVEL_ID                  34722 non-null  object 
 3   VISIT_ORDER                34722 non-null  int64  
 4   VISIT_AREA_NM              34722 non-null  object 
 5   VISIT_START_YMD            34722 non-null  object 
 6   VISIT_END_YMD              34722 non-null  object 
 7   ROAD_NM_ADDR               29537 non-null  object 
 8   LOTNO_ADDR                 34251 non-null  object 
 9   X_COORD                    34487 non-null  float64
 10  Y_COORD                    34487 non-null  float64
 11  ROAD_NM_CD                 0 non-null      float64
 12  LOTNO_CD                   553 non-null    object 
 13  POI_ID                     13740 non-null  obj

In [ ]:
df_1.dropna(subset = ['DGSTFN'], inplace = True)
df_1.reset_index(drop = True, inplace = True)

In [ ]:
df_1

,index,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,...,TRAVEL_STYL_7,TRAVEL_STYL_8,TRAVEL_STATUS_RESIDENCE,TRAVEL_STATUS_DESTINATION,TRAVEL_STATUS_ACCOMPANY,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM
0,0,2208190004,d_d000296,14,산굼부리,2022-08-19,2022-08-19,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
1,1,2208190002,d_d000296,12,한라산국립공원 관음사탐방로,2022-08-19,2022-08-19,NaN,제주특별자치도 제주시 오라이동 산 107-20,126.538006,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
2,2,2208180010,d_d000296,10,한림리조트,2022-08-18,2022-08-19,제주특별자치도 제주시 한림읍 일주서로 5083-5,제주특별자치도 제주시 한림읍 협재리 1193,126.249956,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
3,3,2208200001,d_d000296,17,스누피가든,2022-08-20,2022-08-20,제주특별자치도 제주시 구좌읍 금백조로 916,제주특별자치도 제주시 구좌읍 송당리 2303-1,126.779668,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
4,4,2208180007,d_d000296,7,협재해수욕장,2022-08-18,2022-08-18,제주특별자치도 제주시 한림읍 한림로 329-10,제주특별자치도 제주시 한림읍 협재리 2447,126.239158,...,2,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
34717,38982,2208210008,d_d000621,6,장동민의크라우드pc방,2022-08-21,2022-08-21,제주특별자치도 제주시 건주로4길 10-4,제주특별자치도 제주시 도련일동 1939-3,126.579895,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1
34718,38983,2208230010,d_d000621,28,에코티엘,2022-08-23,2022-08-24,제주특별자치도 제주시 신광로4길 24,제주특별자치도 제주시 연동 251-69,126.487583,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1
34719,38985,2208230009,d_d000621,27,앤유 피시방,2022-08-23,2022-08-23,NaN,제주특별자치도 제주시 연동,NaN,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1
34720,38986,2208230007,d_d000621,25,전주아줌마두루치기,2022-08-23,2022-08-23,제주특별자치도 제주시 신광로4길 25,제주특별자치도 제주시 연동 251-59,126.487416,...,3,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1


In [ ]:
df_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34722 entries, 0 to 34721
Data columns (total 68 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   index                      34722 non-null  int64  
 1   VISIT_AREA_ID              34722 non-null  int64  
 2   TRAVEL_ID                  34722 non-null  object 
 3   VISIT_ORDER                34722 non-null  int64  
 4   VISIT_AREA_NM              34722 non-null  object 
 5   VISIT_START_YMD            34722 non-null  object 
 6   VISIT_END_YMD              34722 non-null  object 
 7   ROAD_NM_ADDR               29537 non-null  object 
 8   LOTNO_ADDR                 34251 non-null  object 
 9   X_COORD                    34487 non-null  float64
 10  Y_COORD                    34487 non-null  float64
 11  ROAD_NM_CD                 0 non-null      float64
 12  LOTNO_CD                   553 non-null    object 
 13  POI_ID                     13740 non-null  obj

- lotno addr 34251 / 34722
lotno 결측치 중 road addr 결측치: 0 / 471

## 3. ROAD NM ADDR, LOTNO ADDR 결측치
ADDR열 만들어서 ROAD NM ADDR 없으면 LOTNO ADDR 사용하고 있으면 그대로 사용

In [ ]:
df_1['LOTNO_ADDR'].iloc[10] == ''

False

In [ ]:
df_1['ADDR'] = ''

In [ ]:
for i in range(len(df_1)):
  if (df_1['ROAD_NM_ADDR'].iloc[i] == '') | (pd.isna(df_1['ROAD_NM_ADDR'].iloc[i])):
    df_1['ADDR'].iloc[i] = df_1['LOTNO_ADDR'].iloc[i]
  else :
    df_1['ADDR'].iloc[i] = df_1['ROAD_NM_ADDR'].iloc[i]

스트리밍 출력 내용이 길어서 마지막 5000줄이 삭제되었습니다.
<ipython-input-77-074b26d1b6db>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_1['ADDR'].iloc[i] = df_1['ROAD_NM_ADDR'].iloc[i]
<ipython-input-77-074b26d1b6db>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_1['ADDR'].iloc[i] = df_1['ROAD_NM_ADDR'].iloc[i]
<ipython-input-77-074b26d1b6db>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_1['ADDR'].iloc[i] = df_1['ROAD_NM_ADDR'].iloc[i]
<ipy

In [ ]:
df_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34722 entries, 0 to 34721
Data columns (total 69 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   index                      34722 non-null  int64  
 1   VISIT_AREA_ID              34722 non-null  int64  
 2   TRAVEL_ID                  34722 non-null  object 
 3   VISIT_ORDER                34722 non-null  int64  
 4   VISIT_AREA_NM              34722 non-null  object 
 5   VISIT_START_YMD            34722 non-null  object 
 6   VISIT_END_YMD              34722 non-null  object 
 7   ROAD_NM_ADDR               29537 non-null  object 
 8   LOTNO_ADDR                 34251 non-null  object 
 9   X_COORD                    34487 non-null  float64
 10  Y_COORD                    34487 non-null  float64
 11  ROAD_NM_CD                 0 non-null      float64
 12  LOTNO_CD                   553 non-null    object 
 13  POI_ID                     13740 non-null  obj

In [ ]:
df_1[df_1['ADDR']=='']

,index,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,...,TRAVEL_STYL_8,TRAVEL_STATUS_RESIDENCE,TRAVEL_STATUS_DESTINATION,TRAVEL_STATUS_ACCOMPANY,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM,ADDR


## 4. 좌표 결측치 해결(X_COORD, Y_COORD)
카카오맵 api(ADDR -> COORD)

In [ ]:
API_KEY = ""

In [ ]:
check = df_1[df_1['X_COORD'].isnull()]
check.info()

<class 'pandas.core.frame.DataFrame'>
Index: 235 entries, 88 to 34719
Data columns (total 69 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   index                      235 non-null    int64  
 1   VISIT_AREA_ID              235 non-null    int64  
 2   TRAVEL_ID                  235 non-null    object 
 3   VISIT_ORDER                235 non-null    int64  
 4   VISIT_AREA_NM              235 non-null    object 
 5   VISIT_START_YMD            235 non-null    object 
 6   VISIT_END_YMD              235 non-null    object 
 7   ROAD_NM_ADDR               218 non-null    object 
 8   LOTNO_ADDR                 24 non-null     object 
 9   X_COORD                    0 non-null      float64
 10  Y_COORD                    0 non-null      float64
 11  ROAD_NM_CD                 0 non-null      float64
 12  LOTNO_CD                   0 non-null      object 
 13  POI_ID                     75 non-null     object 
 

x 좌표 없을 때 y 좌표도 없음(둘 중 하나만 있는 경우는 없음)

In [ ]:
async def fetch_coordinates(session, addr):
    url = f"https://dapi.kakao.com/v2/local/search/address.json?query={addr}"
    headers = {'Authorization': f'KakaoAK {KAKAO_API_KEY}'}
    async with session.get(url, headers=headers) as response:
        if response.status == 200:
            result = await response.json()
            if result['documents']:
                x = result['documents'][0]['address']['x']
                y = result['documents'][0]['address']['y']
                return addr, y, x
        return addr, None, None

async def get_coordinates(df):
    async with aiohttp.ClientSession() as session:
        tasks = []
        for addr in df['ADDR']:
            tasks.append(fetch_coordinates(session, addr))

        results = []
        for task in tqdm.as_completed(tasks, desc='Processing', ncols=100):
            results.append(await task)

        return results

# 데이터 프레임 업데이트
async def update_df(df):
    results = await get_coordinates(df)
    for addr, y, x in results:
        df.loc[df['ADDR'] == addr, ['Y_COORD', 'X_COORD']] = y, x

In [ ]:
# import aiohttp
# import pandas as pd
# from tqdm import tqdm

# GOOGLE_API_KEY = ""

# # 비동기 방식으로 주소를 사용하여 좌표 가져오기
# async def fetch_coordinates(session, addr):
#     url = f"https://maps.googleapis.com/maps/api/geocode/json?address={addr}&key={GOOGLE_API_KEY}"
#     async with session.get(url) as response:
#         if response.status == 200:
#             result = await response.json()
#             if result['results']:
#                 location = result['results'][0]['geometry']['location']
#                 x = location['lng']
#                 y = location['lat']
#                 return addr, y, x
#         return addr, None, None

# # 주소 목록을 사용하여 좌표 가져오기
# async def get_coordinates(df):
#     async with aiohttp.ClientSession() as session:
#         tasks = []
#         for addr in df['ADDR']:
#             tasks.append(fetch_coordinates(session, addr))

#         results = []
#         for task in tqdm(tasks, desc='Processing', ncols=100):
#             results.append(await task)

#         return results

# # 데이터 프레임 업데이트
# async def update_df(df):
#     results = await get_coordinates(df)
#     for addr, y, x in results:
#         df.loc[df['ADDR'] == addr, ['Y_COORD', 'X_COORD']] = y, x


In [ ]:
# 이미 실행 중인 이벤트 루프를 허용하기 위해 필요
nest_asyncio.apply()

In [ ]:
asyncio.run(update_df(check))

Processing: 100%|████████████████████████████████████████████████| 235/235 [00:00<00:00, 241.74it/s]
<ipython-input-164-ae719536b8da>:29: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '33.4260423785235' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df.loc[df['ADDR'] == addr, ['Y_COORD', 'X_COORD']] = y, x
<ipython-input-164-ae719536b8da>:29: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '126.926442688389' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df.loc[df['ADDR'] == addr, ['Y_COORD', 'X_COORD']] = y, x


In [ ]:
check.info()

<class 'pandas.core.frame.DataFrame'>
Index: 235 entries, 88 to 34719
Data columns (total 69 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   index                      235 non-null    int64  
 1   VISIT_AREA_ID              235 non-null    int64  
 2   TRAVEL_ID                  235 non-null    object 
 3   VISIT_ORDER                235 non-null    int64  
 4   VISIT_AREA_NM              235 non-null    object 
 5   VISIT_START_YMD            235 non-null    object 
 6   VISIT_END_YMD              235 non-null    object 
 7   ROAD_NM_ADDR               218 non-null    object 
 8   LOTNO_ADDR                 24 non-null     object 
 9   X_COORD                    191 non-null    object 
 10  Y_COORD                    191 non-null    object 
 11  ROAD_NM_CD                 0 non-null      float64
 12  LOTNO_CD                   0 non-null      object 
 13  POI_ID                     75 non-null     object 
 

In [ ]:
null = check[check['X_COORD'].isnull()]
null

,index,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,...,TRAVEL_STYL_8,TRAVEL_STATUS_RESIDENCE,TRAVEL_STATUS_DESTINATION,TRAVEL_STATUS_ACCOMPANY,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM,ADDR
368,415,2210020003,d_d001585,24,페이보릿,2022-10-02,2022-10-02,제주특별자치도 서귀포시 성산읍 동류암로 20,NaN,None,...,7,경기도,제주,2인 여행(가족 외),2022-09-28~2022-10-04,7,2.0,4.0,1,제주특별자치도 서귀포시 성산읍 동류암로 20
1177,1309,2210230004,d_d006783,3,스위트호텔 제주점,2022-10-23,2022-10-23,제주특별자치도 서귀포시 중문관광로72번길 67,NaN,None,...,4,제주특별자치도,제주,나홀로 여행,2022-10-23~2022-10-24,2,4.0,1.0,0,제주특별자치도 서귀포시 중문관광로72번길 67
1404,1560,2210150002,d_d002839,4,고돌집우럭,2022-10-15,2022-10-15,제주 서귀포시 일주서로 879,NaN,None,...,7,전라북도,제주,자녀 동반 여행,2022-10-15~2022-10-17,3,2.0,5.0,2,제주 서귀포시 일주서로 879
1617,1802,2210150009,d_d002087,9,GS25 이호해수욕장점 건너,2022-10-15,2022-10-15,제주특별자치도 제주시 서해안로 24,NaN,None,...,4,서울특별시,제주,2인 여행(가족 외),2022-10-15~2022-10-19,3,4.0,2.0,1,제주특별자치도 제주시 서해안로 24
1839,2054,2210190005,d_d006124,4,스크루지 펍,2022-10-19,2022-10-19,제주특별자치도 제주시 국기로3길 3,NaN,None,...,6,제주특별자치도,제주,나홀로 여행,2022-10-19~2022-10-20,7,1.0,2.0,0,제주특별자치도 제주시 국기로3길 3
2924,3253,2209280002,d_d001171,10,제주오리엔탈호텔,2022-09-28,2022-09-29,제주특별자치도 제주시 탑동로 47,NaN,None,...,7,충청남도,제주,나홀로 여행,2022-09-28~2022-09-30,1,4.0,6.0,0,제주특별자치도 제주시 탑동로 47
3205,3563,2210080010,d_d004168,27,CU 제주함덕길점,2022-10-08,2022-10-08,제주특별자치도 제주시 조천읍 함덕12길 46-1,NaN,None,...,1,대전광역시,제주,3인 이상 여행(가족 외),2022-10-06~2022-10-09,2,1.0,7.0,2,제주특별자치도 제주시 조천읍 함덕12길 46-1
3675,4086,2210260008,d_d006189,7,온더스톤게스트하우스,2022-10-26,2022-10-26,제주 서귀포시 성산읍 성산등용로 100-4,NaN,None,...,3,경기도,제주,나홀로 여행,2022-10-26~2022-10-29,1,7.0,4.0,0,제주 서귀포시 성산읍 성산등용로 100-4
3676,4087,2210260005,d_d006189,9,온더스톤 게스트하우스,2022-10-26,2022-10-27,제주 서귀포시 성산읍 성산등용로 100-4,NaN,None,...,3,경기도,제주,나홀로 여행,2022-10-26~2022-10-29,1,7.0,4.0,0,제주 서귀포시 성산읍 성산등용로 100-4
4258,4751,2210160007,d_d005437,15,이호정원,2022-10-16,2022-10-16,제주특별자치도 제주시 서해안로 24,NaN,None,...,7,제주특별자치도,제주,2인 가족 여행,2022-10-15~2022-10-16,2,1.0,5.0,1,제주특별자치도 제주시 서해안로 24


In [ ]:
null_check = null[['index', 'VISIT_AREA_NM', 'ROAD_NM_ADDR', 'LOTNO_ADDR', 'ADDR', 'X_COORD', 'Y_COORD']]
null_check

,index,VISIT_AREA_NM,ROAD_NM_ADDR,LOTNO_ADDR,ADDR,X_COORD,Y_COORD
368,415,페이보릿,제주특별자치도 서귀포시 성산읍 동류암로 20,NaN,제주특별자치도 서귀포시 성산읍 동류암로 20,None,None
1177,1309,스위트호텔 제주점,제주특별자치도 서귀포시 중문관광로72번길 67,NaN,제주특별자치도 서귀포시 중문관광로72번길 67,None,None
1404,1560,고돌집우럭,제주 서귀포시 일주서로 879,NaN,제주 서귀포시 일주서로 879,None,None
1617,1802,GS25 이호해수욕장점 건너,제주특별자치도 제주시 서해안로 24,NaN,제주특별자치도 제주시 서해안로 24,None,None
1839,2054,스크루지 펍,제주특별자치도 제주시 국기로3길 3,NaN,제주특별자치도 제주시 국기로3길 3,None,None
2924,3253,제주오리엔탈호텔,제주특별자치도 제주시 탑동로 47,NaN,제주특별자치도 제주시 탑동로 47,None,None
3205,3563,CU 제주함덕길점,제주특별자치도 제주시 조천읍 함덕12길 46-1,NaN,제주특별자치도 제주시 조천읍 함덕12길 46-1,None,None
3675,4086,온더스톤게스트하우스,제주 서귀포시 성산읍 성산등용로 100-4,NaN,제주 서귀포시 성산읍 성산등용로 100-4,None,None
3676,4087,온더스톤 게스트하우스,제주 서귀포시 성산읍 성산등용로 100-4,NaN,제주 서귀포시 성산읍 성산등용로 100-4,None,None
4258,4751,이호정원,제주특별자치도 제주시 서해안로 24,NaN,제주특별자치도 제주시 서해안로 24,None,None


In [ ]:
null_check.info()

<class 'pandas.core.frame.DataFrame'>
Index: 44 entries, 368 to 29712
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   index          44 non-null     int64 
 1   VISIT_AREA_NM  44 non-null     object
 2   ROAD_NM_ADDR   43 non-null     object
 3   LOTNO_ADDR     1 non-null      object
 4   ADDR           44 non-null     object
 5   X_COORD        0 non-null      object
 6   Y_COORD        0 non-null      object
dtypes: int64(1), object(6)
memory usage: 3.8+ KB


In [ ]:
null_check

,index,VISIT_AREA_NM,ROAD_NM_ADDR,LOTNO_ADDR,ADDR,X_COORD,Y_COORD
368,415,페이보릿,제주특별자치도 서귀포시 성산읍 동류암로 20,NaN,제주특별자치도 서귀포시 성산읍 동류암로 20,None,None
1177,1309,스위트호텔 제주점,제주특별자치도 서귀포시 중문관광로72번길 67,NaN,제주특별자치도 서귀포시 중문관광로72번길 67,None,None
1404,1560,고돌집우럭,제주 서귀포시 일주서로 879,NaN,제주 서귀포시 일주서로 879,None,None
1617,1802,GS25 이호해수욕장점 건너,제주특별자치도 제주시 서해안로 24,NaN,제주특별자치도 제주시 서해안로 24,None,None
1839,2054,스크루지 펍,제주특별자치도 제주시 국기로3길 3,NaN,제주특별자치도 제주시 국기로3길 3,None,None
2924,3253,제주오리엔탈호텔,제주특별자치도 제주시 탑동로 47,NaN,제주특별자치도 제주시 탑동로 47,None,None
3205,3563,CU 제주함덕길점,제주특별자치도 제주시 조천읍 함덕12길 46-1,NaN,제주특별자치도 제주시 조천읍 함덕12길 46-1,None,None
3675,4086,온더스톤게스트하우스,제주 서귀포시 성산읍 성산등용로 100-4,NaN,제주 서귀포시 성산읍 성산등용로 100-4,None,None
3676,4087,온더스톤 게스트하우스,제주 서귀포시 성산읍 성산등용로 100-4,NaN,제주 서귀포시 성산읍 성산등용로 100-4,None,None
4258,4751,이호정원,제주특별자치도 제주시 서해안로 24,NaN,제주특별자치도 제주시 서해안로 24,None,None


 구글api, 카카오api로도 처리가 되지 않아 개수가 적어 일일히 찾아서 입력함

In [ ]:
null_df = null_check

In [ ]:
i = 44
print(null_df['VISIT_AREA_NM'].iloc[i])
print(null_df['ADDR'].iloc[i])

IndexError: single positional indexer is out-of-bounds

In [ ]:
# # 특정 행의 열 값을 직접 수정
# null_df.iloc[i, null_df.columns.get_loc('Y_COORD')] = 33.961104
# null_df.iloc[i, null_df.columns.get_loc('X_COORD')] = 126.294812
# null_df.iloc[i]

,29712
index,5348
VISIT_AREA_NM,엘지마트
ROAD_NM_ADDR,제주 제주시 추자면 추자로 70
LOTNO_ADDR,NaN
ADDR,제주 제주시 추자면 추자로 70
X_COORD,126.294812
Y_COORD,33.961104


In [ ]:
null_df

,index,VISIT_AREA_NM,ROAD_NM_ADDR,LOTNO_ADDR,ADDR,X_COORD,Y_COORD
368,5348,페이보릿,제주특별자치도 서귀포시 성산읍 동류암로 20,NaN,제주특별자치도 서귀포시 성산읍 동류암로 20,126.918291,33.450015
1177,5348,스위트호텔 제주점,제주특별자치도 서귀포시 중문관광로72번길 67,NaN,제주특별자치도 서귀포시 중문관광로72번길 67,126.408221,33.249443
1404,5348,고돌집우럭,제주 서귀포시 일주서로 879,NaN,제주 서귀포시 일주서로 879,126.416825,33.25794
1617,5348,GS25 이호해수욕장점 건너,제주특별자치도 제주시 서해안로 24,NaN,제주특별자치도 제주시 서해안로 24,126.456801,33.500078
1839,5348,스크루지 펍,제주특별자치도 제주시 국기로3길 3,NaN,제주특별자치도 제주시 국기로3길 3,126.490845,33.478792
2924,5348,제주오리엔탈호텔,제주특별자치도 제주시 탑동로 47,NaN,제주특별자치도 제주시 탑동로 47,126.519926,33.517264
3205,5348,CU 제주함덕길점,제주특별자치도 제주시 조천읍 함덕12길 46-1,NaN,제주특별자치도 제주시 조천읍 함덕12길 46-1,126.665109,33.537981
3675,5348,온더스톤게스트하우스,제주 서귀포시 성산읍 성산등용로 100-4,NaN,제주 서귀포시 성산읍 성산등용로 100-4,126.930621,33.470146
3676,5348,온더스톤 게스트하우스,제주 서귀포시 성산읍 성산등용로 100-4,NaN,제주 서귀포시 성산읍 성산등용로 100-4,126.930621,33.470146
4258,5348,이호정원,제주특별자치도 제주시 서해안로 24,NaN,제주특별자치도 제주시 서해안로 24,126.456959,33.4999


In [ ]:
null

,index,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,...,TRAVEL_STYL_8,TRAVEL_STATUS_RESIDENCE,TRAVEL_STATUS_DESTINATION,TRAVEL_STATUS_ACCOMPANY,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM,ADDR
368,415,2210020003,d_d001585,24,페이보릿,2022-10-02,2022-10-02,제주특별자치도 서귀포시 성산읍 동류암로 20,NaN,None,...,7,경기도,제주,2인 여행(가족 외),2022-09-28~2022-10-04,7,2.0,4.0,1,제주특별자치도 서귀포시 성산읍 동류암로 20
1177,1309,2210230004,d_d006783,3,스위트호텔 제주점,2022-10-23,2022-10-23,제주특별자치도 서귀포시 중문관광로72번길 67,NaN,None,...,4,제주특별자치도,제주,나홀로 여행,2022-10-23~2022-10-24,2,4.0,1.0,0,제주특별자치도 서귀포시 중문관광로72번길 67
1404,1560,2210150002,d_d002839,4,고돌집우럭,2022-10-15,2022-10-15,제주 서귀포시 일주서로 879,NaN,None,...,7,전라북도,제주,자녀 동반 여행,2022-10-15~2022-10-17,3,2.0,5.0,2,제주 서귀포시 일주서로 879
1617,1802,2210150009,d_d002087,9,GS25 이호해수욕장점 건너,2022-10-15,2022-10-15,제주특별자치도 제주시 서해안로 24,NaN,None,...,4,서울특별시,제주,2인 여행(가족 외),2022-10-15~2022-10-19,3,4.0,2.0,1,제주특별자치도 제주시 서해안로 24
1839,2054,2210190005,d_d006124,4,스크루지 펍,2022-10-19,2022-10-19,제주특별자치도 제주시 국기로3길 3,NaN,None,...,6,제주특별자치도,제주,나홀로 여행,2022-10-19~2022-10-20,7,1.0,2.0,0,제주특별자치도 제주시 국기로3길 3
2924,3253,2209280002,d_d001171,10,제주오리엔탈호텔,2022-09-28,2022-09-29,제주특별자치도 제주시 탑동로 47,NaN,None,...,7,충청남도,제주,나홀로 여행,2022-09-28~2022-09-30,1,4.0,6.0,0,제주특별자치도 제주시 탑동로 47
3205,3563,2210080010,d_d004168,27,CU 제주함덕길점,2022-10-08,2022-10-08,제주특별자치도 제주시 조천읍 함덕12길 46-1,NaN,None,...,1,대전광역시,제주,3인 이상 여행(가족 외),2022-10-06~2022-10-09,2,1.0,7.0,2,제주특별자치도 제주시 조천읍 함덕12길 46-1
3675,4086,2210260008,d_d006189,7,온더스톤게스트하우스,2022-10-26,2022-10-26,제주 서귀포시 성산읍 성산등용로 100-4,NaN,None,...,3,경기도,제주,나홀로 여행,2022-10-26~2022-10-29,1,7.0,4.0,0,제주 서귀포시 성산읍 성산등용로 100-4
3676,4087,2210260005,d_d006189,9,온더스톤 게스트하우스,2022-10-26,2022-10-27,제주 서귀포시 성산읍 성산등용로 100-4,NaN,None,...,3,경기도,제주,나홀로 여행,2022-10-26~2022-10-29,1,7.0,4.0,0,제주 서귀포시 성산읍 성산등용로 100-4
4258,4751,2210160007,d_d005437,15,이호정원,2022-10-16,2022-10-16,제주특별자치도 제주시 서해안로 24,NaN,None,...,7,제주특별자치도,제주,2인 가족 여행,2022-10-15~2022-10-16,2,1.0,5.0,1,제주특별자치도 제주시 서해안로 24


In [ ]:
check_save = check

In [ ]:
check_save.info()

ERROR:asyncio:Task exception was never retrieved
future: <Task finished name='Task-1' coro=<update_df() done, defined at <ipython-input-49-a97029ce816f>:26> exception=KeyboardInterrupt()>
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py", line 3553, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-52-9733aae39860>", line 1, in <cell line: 1>
    asyncio.run(update_df(df))
  File "/usr/local/lib/python3.10/dist-packages/nest_asyncio.py", line 30, in run
    return loop.run_until_complete(task)
  File "/usr/local/lib/python3.10/dist-packages/nest_asyncio.py", line 92, in run_until_complete
    self._run_once()
  File "/usr/local/lib/python3.10/dist-packages/nest_asyncio.py", line 133, in _run_once
    handle._run()
  File "/usr/lib/python3.10/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.10/asyncio/tasks.py", line 23

<class 'pandas.core.frame.DataFrame'>
Index: 235 entries, 88 to 34719
Data columns (total 69 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   index                      235 non-null    int64  
 1   VISIT_AREA_ID              235 non-null    int64  
 2   TRAVEL_ID                  235 non-null    object 
 3   VISIT_ORDER                235 non-null    int64  
 4   VISIT_AREA_NM              235 non-null    object 
 5   VISIT_START_YMD            235 non-null    object 
 6   VISIT_END_YMD              235 non-null    object 
 7   ROAD_NM_ADDR               218 non-null    object 
 8   LOTNO_ADDR                 24 non-null     object 
 9   X_COORD                    191 non-null    object 
 10  Y_COORD                    191 non-null    object 
 11  ROAD_NM_CD                 0 non-null      float64
 12  LOTNO_CD                   0 non-null      object 
 13  POI_ID                     75 non-null     object 
 

In [ ]:
check.update(null_df)

In [ ]:
df_1.update(check)

In [ ]:
df_1.dropna(subset=['X_COORD'], inplace = True)

In [ ]:
df_1.reset_index(drop = True, inplace = True)
df_1

,index,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,...,TRAVEL_STYL_8,TRAVEL_STATUS_RESIDENCE,TRAVEL_STATUS_DESTINATION,TRAVEL_STATUS_ACCOMPANY,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM,ADDR
0,0,2208190004,d_d000296,14,산굼부리,2022-08-19,2022-08-19,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 조천읍 교래리 166-2
1,1,2208190002,d_d000296,12,한라산국립공원 관음사탐방로,2022-08-19,2022-08-19,NaN,제주특별자치도 제주시 오라이동 산 107-20,126.538006,...,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 오라이동 산 107-20
2,2,2208180010,d_d000296,10,한림리조트,2022-08-18,2022-08-19,제주특별자치도 제주시 한림읍 일주서로 5083-5,제주특별자치도 제주시 한림읍 협재리 1193,126.249956,...,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 한림읍 일주서로 5083-5
3,3,2208200001,d_d000296,17,스누피가든,2022-08-20,2022-08-20,제주특별자치도 제주시 구좌읍 금백조로 916,제주특별자치도 제주시 구좌읍 송당리 2303-1,126.779668,...,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 구좌읍 금백조로 916
4,4,2208180007,d_d000296,7,협재해수욕장,2022-08-18,2022-08-18,제주특별자치도 제주시 한림읍 한림로 329-10,제주특별자치도 제주시 한림읍 협재리 2447,126.239158,...,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 한림읍 한림로 329-10
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
34715,38982,2208210008,d_d000621,6,장동민의크라우드pc방,2022-08-21,2022-08-21,제주특별자치도 제주시 건주로4길 10-4,제주특별자치도 제주시 도련일동 1939-3,126.579895,...,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 건주로4길 10-4
34716,38983,2208230010,d_d000621,28,에코티엘,2022-08-23,2022-08-24,제주특별자치도 제주시 신광로4길 24,제주특별자치도 제주시 연동 251-69,126.487583,...,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 24
34717,38985,2208230009,d_d000621,27,앤유 피시방,2022-08-23,2022-08-23,NaN,제주특별자치도 제주시 연동,126.496886364512,...,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 연동
34718,38986,2208230007,d_d000621,25,전주아줌마두루치기,2022-08-23,2022-08-23,제주특별자치도 제주시 신광로4길 25,제주특별자치도 제주시 연동 251-59,126.487416,...,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 25


In [ ]:
df_1.columns

Index(['index', 'VISIT_AREA_ID', 'TRAVEL_ID', 'VISIT_ORDER', 'VISIT_AREA_NM',
       'VISIT_START_YMD', 'VISIT_END_YMD', 'ROAD_NM_ADDR', 'LOTNO_ADDR',
       'X_COORD', 'Y_COORD', 'ROAD_NM_CD', 'LOTNO_CD', 'POI_ID', 'POI_NM',
       'RESIDENCE_TIME_MIN', 'VISIT_AREA_TYPE_CD', 'REVISIT_YN',
       'VISIT_CHC_REASON_CD', 'LODGING_TYPE_CD', 'DGSTFN', 'REVISIT_INTENTION',
       'RCMDTN_INTENTION', 'SGG_CD', 'TRAVEL_NM', 'TRAVELER_ID',
       'TRAVEL_PURPOSE', 'TRAVEL_START_YMD', 'TRAVEL_END_YMD', 'MVMN_NM',
       'TRAVEL_MISSION', 'TRAVEL_MISSION_CHECK', 'TRAVEL_MISSION_PRIORITY',
       'RESIDENCE_SGG_CD', 'GENDER', 'AGE_GRP', 'EDU_NM', 'EDU_FNSH_SE',
       'MARR_STTS', 'FAMILY_MEMB', 'JOB_NM', 'JOB_ETC', 'INCOME',
       'HOUSE_INCOME', 'TRAVEL_TERM', 'TRAVEL_NUM', 'TRAVEL_LIKE_SIDO_1',
       'TRAVEL_LIKE_SGG_1', 'TRAVEL_LIKE_SIDO_2', 'TRAVEL_LIKE_SGG_2',
       'TRAVEL_LIKE_SIDO_3', 'TRAVEL_LIKE_SGG_3', 'TRAVEL_STYL_1',
       'TRAVEL_STYL_2', 'TRAVEL_STYL_3', 'TRAVEL_STYL_4', 'TRAV

In [ ]:
df_1.drop(columns = ['index'], inplace = True)
df_1

,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,Y_COORD,...,TRAVEL_STYL_8,TRAVEL_STATUS_RESIDENCE,TRAVEL_STATUS_DESTINATION,TRAVEL_STATUS_ACCOMPANY,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM,ADDR
0,2208190004,d_d000296,14,산굼부리,2022-08-19,2022-08-19,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,33.432574,...,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 조천읍 교래리 166-2
1,2208190002,d_d000296,12,한라산국립공원 관음사탐방로,2022-08-19,2022-08-19,NaN,제주특별자치도 제주시 오라이동 산 107-20,126.538006,33.390838,...,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 오라이동 산 107-20
2,2208180010,d_d000296,10,한림리조트,2022-08-18,2022-08-19,제주특별자치도 제주시 한림읍 일주서로 5083-5,제주특별자치도 제주시 한림읍 협재리 1193,126.249956,33.389418,...,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 한림읍 일주서로 5083-5
3,2208200001,d_d000296,17,스누피가든,2022-08-20,2022-08-20,제주특별자치도 제주시 구좌읍 금백조로 916,제주특별자치도 제주시 구좌읍 송당리 2303-1,126.779668,33.443359,...,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 구좌읍 금백조로 916
4,2208180007,d_d000296,7,협재해수욕장,2022-08-18,2022-08-18,제주특별자치도 제주시 한림읍 한림로 329-10,제주특별자치도 제주시 한림읍 협재리 2447,126.239158,33.393866,...,5,서울특별시,제주,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 한림읍 한림로 329-10
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
34715,2208210008,d_d000621,6,장동민의크라우드pc방,2022-08-21,2022-08-21,제주특별자치도 제주시 건주로4길 10-4,제주특별자치도 제주시 도련일동 1939-3,126.579895,33.516333,...,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 건주로4길 10-4
34716,2208230010,d_d000621,28,에코티엘,2022-08-23,2022-08-24,제주특별자치도 제주시 신광로4길 24,제주특별자치도 제주시 연동 251-69,126.487583,33.489267,...,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 24
34717,2208230009,d_d000621,27,앤유 피시방,2022-08-23,2022-08-23,NaN,제주특별자치도 제주시 연동,126.496886364512,33.4881587278291,...,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 연동
34718,2208230007,d_d000621,25,전주아줌마두루치기,2022-08-23,2022-08-23,제주특별자치도 제주시 신광로4길 25,제주특별자치도 제주시 연동 251-59,126.487416,33.488756,...,5,경기도,제주,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 25


## 5. 방문지 대표 유형 선정(VISIT_AREA_MAIN_TYPE)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34720 entries, 0 to 34719
Data columns (total 70 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   VISIT_AREA_ID              34720 non-null  int64  
 1   TRAVEL_ID                  34720 non-null  object 
 2   VISIT_ORDER                34720 non-null  int64  
 3   VISIT_AREA_NM              34720 non-null  object 
 4   VISIT_START_YMD            34720 non-null  object 
 5   VISIT_END_YMD              34720 non-null  object 
 6   ROAD_NM_ADDR               29536 non-null  object 
 7   LOTNO_ADDR                 34250 non-null  object 
 8   X_COORD                    34720 non-null  float64
 9   Y_COORD                    34720 non-null  float64
 10  ROAD_NM_CD                 0 non-null      float64
 11  LOTNO_CD                   0 non-null      float64
 12  POI_ID                     13740 non-null  object 
 13  POI_NM                     13740 non-null  obj

In [ ]:
# 중복 제거 및 인덱스 초기화
place = df[['VISIT_AREA_NM', 'VISIT_AREA_TYPE_CD']]
place.reset_index(drop=True, inplace=True)

In [ ]:
place

,VISIT_AREA_NM,VISIT_AREA_TYPE_CD
0,산굼부리,1
1,한라산국립공원 관음사탐방로,1
2,한림리조트,24
3,스누피가든,6
4,협재해수욕장,1
...,...,...
34715,장동민의크라우드pc방,3
34716,에코티엘,24
34717,앤유 피시방,3
34718,전주아줌마두루치기,11


In [ ]:
# VISIT_AREA_NM에 따른 VISIT_AREA_TYPE_CD의 최빈값 찾기
main_type = place.groupby('VISIT_AREA_NM')['VISIT_AREA_TYPE_CD'].agg(lambda x: x.mode()[0]).reset_index()

# 열 이름 변경
main_type.rename(columns={'VISIT_AREA_TYPE_CD': 'VISIT_AREA_MAIN_TYPE'}, inplace=True)

# VISIT_AREA_MAIN_TYPE 열 추가
place = place.merge(main_type, on='VISIT_AREA_NM')

In [ ]:
place

,VISIT_AREA_NM,VISIT_AREA_TYPE_CD,VISIT_AREA_MAIN_TYPE
0,산굼부리,1,1
1,산굼부리,1,1
2,산굼부리,1,1
3,산굼부리,1,1
4,산굼부리,2,1
...,...,...,...
34715,미니K코인노래연습장,4,4
34716,장동민의크라우드pc방,3,3
34717,앤유 피시방,3,3
34718,전주아줌마두루치기,11,11


In [ ]:
merged = pd.merge(df, place, on = ['VISIT_AREA_NM', 'VISIT_AREA_TYPE_CD'], how = 'left')

In [ ]:
merged

,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,Y_COORD,...,TRAVEL_STATUS_ACCOMPANY,TRAVEL_STATUS_YMD,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM,ADDR,RES_CAFE,VISIT_COUNT,VISIT_AREA_MAIN_TYPE
0,2208190004,d_d000296,14,산굼부리,2022-08-19 00:00:00,2022-08-19 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,33.432574,...,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1
1,2208190004,d_d000296,14,산굼부리,2022-08-19 00:00:00,2022-08-19 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,33.432574,...,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1
2,2208190004,d_d000296,14,산굼부리,2022-08-19 00:00:00,2022-08-19 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,33.432574,...,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1
3,2208190004,d_d000296,14,산굼부리,2022-08-19 00:00:00,2022-08-19 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,33.432574,...,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1
4,2208190004,d_d000296,14,산굼부리,2022-08-19 00:00:00,2022-08-19 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,33.432574,...,자녀 동반 여행,2022-08-19~2022-08-21,3,7.0,1.0,3,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1300213,2208230010,d_d000621,28,에코티엘,2022-08-23 00:00:00,2022-08-24 00:00:00,제주특별자치도 제주시 신광로4길 24,제주특별자치도 제주시 연동 251-69,126.487583,33.489267,...,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 24,2,9,24
1300214,2208230010,d_d000621,28,에코티엘,2022-08-23 00:00:00,2022-08-24 00:00:00,제주특별자치도 제주시 신광로4길 24,제주특별자치도 제주시 연동 251-69,126.487583,33.489267,...,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 24,2,9,24
1300215,2208230009,d_d000621,27,앤유 피시방,2022-08-23 00:00:00,2022-08-23 00:00:00,NaN,제주특별자치도 제주시 연동,126.496886,33.488159,...,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 연동,2,1,3
1300216,2208230007,d_d000621,25,전주아줌마두루치기,2022-08-23 00:00:00,2022-08-23 00:00:00,제주특별자치도 제주시 신광로4길 25,제주특별자치도 제주시 연동 251-59,126.487416,33.488756,...,2인 가족 여행,2022-08-21~2022-08-24,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 25,0,1,11


In [ ]:
merged.drop_duplicates(inplace = True)
merged.info()

<class 'pandas.core.frame.DataFrame'>
Index: 34720 entries, 0 to 1300217
Data columns (total 71 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   VISIT_AREA_ID              34720 non-null  int64  
 1   TRAVEL_ID                  34720 non-null  object 
 2   VISIT_ORDER                34720 non-null  int64  
 3   VISIT_AREA_NM              34720 non-null  object 
 4   VISIT_START_YMD            34720 non-null  object 
 5   VISIT_END_YMD              34720 non-null  object 
 6   ROAD_NM_ADDR               29536 non-null  object 
 7   LOTNO_ADDR                 34250 non-null  object 
 8   X_COORD                    34720 non-null  float64
 9   Y_COORD                    34720 non-null  float64
 10  ROAD_NM_CD                 0 non-null      float64
 11  LOTNO_CD                   0 non-null      float64
 12  POI_ID                     13740 non-null  object 
 13  POI_NM                     13740 non-null  object

In [ ]:
merged.to_csv("Data/Final/final_01.csv")

## 방문 횟수 계산

In [ ]:
# 여행지별 총 방문 횟수를 계산
total_visits = df.groupby('VISIT_AREA_NM').size().reset_index(name='VISIT_COUNT')

# 원래 데이터프레임에 병합
df = df.merge(total_visits, on='VISIT_AREA_NM', how='left')

In [ ]:
df.info(

)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34720 entries, 0 to 34719
Data columns (total 72 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Unnamed: 0                 34720 non-null  int64  
 1   VISIT_AREA_ID              34720 non-null  int64  
 2   TRAVEL_ID                  34720 non-null  object 
 3   VISIT_ORDER                34720 non-null  int64  
 4   VISIT_AREA_NM              34720 non-null  object 
 5   VISIT_START_YMD            34720 non-null  object 
 6   VISIT_END_YMD              34720 non-null  object 
 7   ROAD_NM_ADDR               29536 non-null  object 
 8   LOTNO_ADDR                 34250 non-null  object 
 9   X_COORD                    34720 non-null  float64
 10  Y_COORD                    34720 non-null  float64
 11  ROAD_NM_CD                 0 non-null      float64
 12  LOTNO_CD                   0 non-null      float64
 13  POI_ID                     13740 non-null  obj

## 평균 체류시간 생성

In [ ]:
# 방문지별 평균 체류 시간 계산
average_time = df.groupby('VISIT_AREA_NM')['RESIDENCE_TIME_MIN'].mean().reset_index()

# 컬럼 이름 변경
average_time.columns = ['VISIT_AREA_NM', 'AVG_TIME_MIN']

# 평균 시간을 10분 단위로 레벨링
average_time['AVG_TIME_MIN'] = (average_time['AVG_TIME_MIN'] // 10) * 10

# 원래 데이터 프레임에 평균 체류 시간 데이터를 추가 (merge 사용)
df = pd.merge(df, average_time, on='VISIT_AREA_NM', how='left')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34720 entries, 0 to 34719
Data columns (total 73 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Unnamed: 0                 34720 non-null  int64  
 1   VISIT_AREA_ID              34720 non-null  int64  
 2   TRAVEL_ID                  34720 non-null  object 
 3   VISIT_ORDER                34720 non-null  int64  
 4   VISIT_AREA_NM              34720 non-null  object 
 5   VISIT_START_YMD            34720 non-null  object 
 6   VISIT_END_YMD              34720 non-null  object 
 7   ROAD_NM_ADDR               29536 non-null  object 
 8   LOTNO_ADDR                 34250 non-null  object 
 9   X_COORD                    34720 non-null  float64
 10  Y_COORD                    34720 non-null  float64
 11  ROAD_NM_CD                 0 non-null      float64
 12  LOTNO_CD                   0 non-null      float64
 13  POI_ID                     13740 non-null  obj

## 활동 소비내역

In [ ]:
consume = pd.read_csv("원본 데이터/Merge/Consume_Data.csv")
consume

,TRAVEL_ID,VISIT_AREA_ID,ACTIVITY_TYPE_CD,ACTIVITY_TYPE_SEQ,CONSUME_HIS_SEQ,CONSUME_HIS_SNO,PAYMENT_NUM,BRNO,STORE_NM,ROAD_NM_ADDR,LOTNO_ADDR,ROAD_NM_CD,LOTNO_CD,PAYMENT_DT,PAYMENT_MTHD_SE,PAYMENT_AMT_WON,PAYMENT_ETC,SGG_CD
0,d_d010266,2211070007,1,0,0,1,1,2.018122e+09,주식회사 스타벅스 커피 코리아,제주특별자치도 제주시 애월읍 애월해안로 376,제주특별자치도 제주시 애월읍 고내리 340,3349131.0,5.011025e+09,2022-11-07 17:08:00.000,3.0,6800.0,G)제주F.콜드브루,NaN
1,d_d000281,2208190005,1,0,0,1,4,8.247700e+09,갈치옥,제주특별자치도 제주시 조천읍 조함해안로 530,제주특별자치도 제주시 조천읍 함덕리 1004-30,3349182.0,5.011026e+09,2022-08-19 14:22:00.000,1.0,109000.0,통갈치구이세트(4인),NaN
2,d_d003962,2210140003,1,0,0,1,6,4.620801e+09,은희네해장국성산점,제주특별자치도 서귀포시 성산읍 환해장성로 803,제주특별자치도 서귀포시 성산읍 온평리 1-2,3350332.0,5.013026e+09,2022-10-14 08:21:00.000,1.0,40000.0,소고기해장국,NaN
3,d_d004673,2210230002,1,0,0,1,2,6.162985e+09,유동커피,제주특별자치도 서귀포시 태평로 406-2,제주특별자치도 서귀포시 서귀동 581-4,3350269.0,5.013010e+09,2022-10-23 12:43:00.000,1.0,13900.0,ICE아메리카노A;메이플크로플;ICE카페비엔나블,NaN
4,d_d005572,2210140008,2,0,0,1,1,8.338600e+09,홀라인,제주특별자치도 제주시 구좌읍 평대2길 34-1,제주특별자치도 제주시 구좌읍 평대리 1984-1,4848921.0,5.011026e+09,2022-10-14 16:42:00.000,1.0,63900.0,(에이엠지티타늄)티타늄집게;(제로그램)21비스트로수저세트;(선데이서비스클럽)주전자꼭지,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
38015,d_d000150,2208140008,3,0,0,1,1,2.212751e+09,금능석물원,제주특별자치도 제주시 한림읍 한림로 176,제주특별자치도 제주시 한림읍 금능리 1282-9,3349208.0,5.011025e+09,2022-08-14 14:00:00.000,1.0,4000.0,성인(20세이상,NaN
38016,d_d004817,2210270015,1,0,0,1,2,1.051946e+09,콜럼버스,제주특별자치도 제주시 구좌읍 세화4길 10,제주특별자치도 제주시 구좌읍 세화리 1448-1,4848012.0,5.011026e+09,2022-10-27 19:12:00.000,1.0,132000.0,배꼽살&대뱃살;풍정사계-춘;오믈렛 국물떡볶이,NaN
38017,b_b000051,2209100007,1,4,4,1,4,6.120184e+09,대전광봉횟집,경남 통영시 미수해안로 120,경남 통영시 미수동 8-1,3333009.0,4.822012e+09,2022-09-10 20:24:00.000,1.0,50000.0,NaN,NaN
38018,d_d000431,2208240003,1,1,1,1,6,6.163059e+09,씨유협재제천점,제주특별자치도 제주시 한림읍 한림로 329-8,제주특별자치도 제주시 한림읍 협재리 2447-43,3349208.0,5.011025e+09,2022-08-24 15:46:00.000,1.0,15800.0,탱글탱글샤인머스캣;자일리톨F빅팩껌;토레타P900ML;더하다옥수수P500ML;자연은제...,NaN


In [ ]:
pay = consume[['TRAVEL_ID', 'VISIT_AREA_ID','PAYMENT_NUM', 'PAYMENT_AMT_WON']]
pay.drop_duplicates(inplace = True)
pay.reset_index(drop = True, inplace = True)
pay.dropna(inplace = True)
pay.info()

<class 'pandas.core.frame.DataFrame'>
Index: 37671 entries, 0 to 37682
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   TRAVEL_ID        37671 non-null  object 
 1   VISIT_AREA_ID    37671 non-null  int64  
 2   PAYMENT_NUM      37671 non-null  int64  
 3   PAYMENT_AMT_WON  37671 non-null  float64
dtypes: float64(1), int64(2), object(1)
memory usage: 1.4+ MB


<ipython-input-35-82c655d9910d>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  pay.drop_duplicates(inplace = True)
<ipython-input-35-82c655d9910d>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  pay.dropna(inplace = True)


In [ ]:
check = pd.merge(df, pay, on = ['TRAVEL_ID', 'VISIT_AREA_ID'], how = 'left')
check.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 37472 entries, 0 to 37471
Data columns (total 75 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Unnamed: 0                 37472 non-null  int64  
 1   VISIT_AREA_ID              37472 non-null  int64  
 2   TRAVEL_ID                  37472 non-null  object 
 3   VISIT_ORDER                37472 non-null  int64  
 4   VISIT_AREA_NM              37472 non-null  object 
 5   VISIT_START_YMD            37472 non-null  object 
 6   VISIT_END_YMD              37472 non-null  object 
 7   ROAD_NM_ADDR               32028 non-null  object 
 8   LOTNO_ADDR                 36959 non-null  object 
 9   X_COORD                    37472 non-null  float64
 10  Y_COORD                    37472 non-null  float64
 11  ROAD_NM_CD                 0 non-null      float64
 12  LOTNO_CD                   0 non-null      float64
 13  POI_ID                     15279 non-null  obj

In [ ]:
check_1 = check[['VISIT_AREA_NM', 'VISIT_AREA_MAIN_TYPE','PAYMENT_NUM', 'PAYMENT_AMT_WON']]
check_1.drop_duplicates(inplace = True)
check_1.reset_index(drop = True, inplace = True)
check_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21793 entries, 0 to 21792
Data columns (total 4 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   VISIT_AREA_NM         21793 non-null  object 
 1   VISIT_AREA_MAIN_TYPE  21793 non-null  int64  
 2   PAYMENT_NUM           17680 non-null  float64
 3   PAYMENT_AMT_WON       17680 non-null  float64
dtypes: float64(2), int64(1), object(1)
memory usage: 681.2+ KB


<ipython-input-63-be4d53ebd7b2>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  check_1.drop_duplicates(inplace = True)


In [ ]:
check_1['PAYMENT_PER_PS'] = 0

<ipython-input-64-c75b93ec1ca3>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  check_1['PAYMENT_PER_PS'] = 0


In [ ]:
check_1

,VISIT_AREA_NM,VISIT_AREA_MAIN_TYPE,PAYMENT_NUM,PAYMENT_AMT_WON,PAYMENT_PER_PS
0,산굼부리,1,4.0,0.0,0
1,산굼부리,1,4.0,17665.0,0
2,한라산국립공원 관음사탐방로,1,NaN,NaN,0
3,한림리조트,24,NaN,NaN,0
4,스누피가든,6,4.0,46500.0,0
...,...,...,...,...,...
21788,장동민의크라우드pc방,3,2.0,10000.0,0
21789,장동민의크라우드pc방,3,2.0,13500.0,0
21790,앤유 피시방,3,NaN,NaN,0
21791,전주아줌마두루치기,11,NaN,NaN,0


In [ ]:
check_1['PAYMENT_PER_PS'] = check['PAYMENT_AMT_WON'] / check['PAYMENT_NUM']
check_1

<ipython-input-66-f5cca867b579>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  check_1['PAYMENT_PER_PS'] = check['PAYMENT_AMT_WON'] / check['PAYMENT_NUM']


,VISIT_AREA_NM,VISIT_AREA_MAIN_TYPE,PAYMENT_NUM,PAYMENT_AMT_WON,PAYMENT_PER_PS
0,산굼부리,1,4.0,0.0,0.00
1,산굼부리,1,4.0,17665.0,4416.25
2,한라산국립공원 관음사탐방로,1,NaN,NaN,NaN
3,한림리조트,24,NaN,NaN,NaN
4,스누피가든,6,4.0,46500.0,11625.00
...,...,...,...,...,...
21788,장동민의크라우드pc방,3,2.0,10000.0,19500.00
21789,장동민의크라우드pc방,3,2.0,13500.0,7000.00
21790,앤유 피시방,3,NaN,NaN,NaN
21791,전주아줌마두루치기,11,NaN,NaN,14000.00


In [ ]:
# 그룹화하여 평균 계산 _ 소수 2번째 자리까지
avg_check = check_1.groupby("VISIT_AREA_NM")['PAYMENT_PER_PS'].mean().round(2)

# 데이터프레임 생성
avg_df = pd.DataFrame(avg_check).reset_index()

# 컬럼 이름 지정
avg_df.columns = ['VISIT_AREA_NM', 'PAYMENT_AVG']

avg_df

,VISIT_AREA_NM,PAYMENT_AVG
0,064노트,15475.00
1,064바당,750.00
2,064푸드트럭,NaN
3,104스테이,6750.00
4,10호어린이공원,5000.00
...,...,...
7593,히츠야,4166.67
7594,히포파운드,8666.67
7595,힐록,9000.00
7596,힐링스테이,4500.00


In [ ]:
# 기존의 데이터 프레임에 열 추가
place = pd.merge(check_1, avg_df, on='VISIT_AREA_NM', how = 'left')

pay = place[['VISIT_AREA_NM', 'VISIT_AREA_MAIN_TYPE', 'PAYMENT_AVG']].drop_duplicates()

pay.reset_index(drop = True, inplace = True)

pay_check = pay[pay['VISIT_AREA_MAIN_TYPE'] == 11]
pay_check.drop_duplicates(inplace = True)
pay_check.reset_index(drop = True, inplace = True)


pay_check[pay_check['PAYMENT_AVG'].isna()]

<ipython-input-86-83216e731fc0>:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  pay_check.drop_duplicates(inplace = True)


,VISIT_AREA_NM,VISIT_AREA_MAIN_TYPE,PAYMENT_AVG
26,컴포즈커피 서귀포서홍점,11,NaN
27,비지곶식당,11,NaN
35,과물치킨,11,NaN
49,엘리펀트힙,11,NaN
50,일도가공,11,NaN
...,...,...,...
4095,멜버른32,11,NaN
4101,편백돈나무,11,NaN
4102,섬돼지,11,NaN
4109,곰돌식당,11,NaN


In [ ]:
pay_check.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4112 entries, 0 to 4111
Data columns (total 3 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   VISIT_AREA_NM         4112 non-null   object 
 1   VISIT_AREA_MAIN_TYPE  4112 non-null   int64  
 2   PAYMENT_AVG           2938 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 96.5+ KB


In [ ]:
pay.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7598 entries, 0 to 7597
Data columns (total 3 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   VISIT_AREA_NM         7598 non-null   object 
 1   VISIT_AREA_MAIN_TYPE  7598 non-null   int64  
 2   PAYMENT_AVG           5307 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 178.2+ KB


## 방문 선택 이유

In [ ]:
reason_df = df[['VISIT_AREA_NM', 'VISIT_CHC_REASON_CD']]
reason_df

,VISIT_AREA_NM,VISIT_CHC_REASON_CD
0,산굼부리,4
1,한라산국립공원 관음사탐방로,1
2,한림리조트,8
3,스누피가든,1
4,협재해수욕장,1
...,...,...
34715,장동민의크라우드pc방,9
34716,에코티엘,11
34717,앤유 피시방,4
34718,전주아줌마두루치기,9


In [ ]:
reason_df[reason_df['VISIT_AREA_NM'] == '한림리조트'].value_counts()

VISIT_AREA_NM  VISIT_CHC_REASON_CD
한림리조트          11                     8
               8                      6
               2                      4
               7                      1
Name: count, dtype: int64

In [ ]:
# VISIT_AREA_NM에 따른 VISIT_AREA_TYPE_CD의 최빈값 찾기
main_reason = reason_df.groupby('VISIT_AREA_NM')['VISIT_CHC_REASON_CD'].agg(lambda x: x.mode()[0]).reset_index()

# 열 이름 변경
main_reason.rename(columns={'VISIT_CHC_REASON_CD': 'VISIT_MAIN_REASON'}, inplace=True)

# VISIT_AREA_MAIN_TYPE 열 추가
df = df.merge(main_reason, on='VISIT_AREA_NM')

In [ ]:
df

,Unnamed: 0,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,...,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM,ADDR,RES_CAFE,VISIT_COUNT,VISIT_AREA_MAIN_TYPE,AVG_TIME_MIN,VISIT_MAIN_REASON
0,0,2208190004,d_d000296,14,산굼부리,2022-08-19 00:00:00,2022-08-19 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,3,7.0,1.0,3,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1,60.0,1
1,3658,2210170005,d_d003259,4,산굼부리,2022-10-17 00:00:00,2022-10-17 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,2,1.0,3.0,1,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1,60.0,1
2,9721,2209280002,d_d001522,21,산굼부리,2022-09-28 00:00:00,2022-09-28 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,2,3.0,1.0,5,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1,60.0,1
3,28034,2211010002,d_d010245,22,산굼부리,2022-11-01 00:00:00,2022-11-01 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,1,2.0,6.0,1,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1,60.0,1
4,51305,2210240001,d_d003054,8,산굼부리,2022-10-24 00:00:00,2022-10-24 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,3,2.0,5.0,1,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1,60.0,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
34715,1300145,2208210007,d_d000621,5,미니K코인노래연습장,2022-08-21 00:00:00,2022-08-21 00:00:00,제주특별자치도 제주시 국기로 5,제주특별자치도 제주시 연동 1457,126.490036,...,1,2.0,7.0,1,제주특별자치도 제주시 국기로 5,2,1,4,30.0,9
34716,1300205,2208210008,d_d000621,6,장동민의크라우드pc방,2022-08-21 00:00:00,2022-08-21 00:00:00,제주특별자치도 제주시 건주로4길 10-4,제주특별자치도 제주시 도련일동 1939-3,126.579895,...,1,2.0,7.0,1,제주특별자치도 제주시 건주로4길 10-4,2,1,3,150.0,9
34717,1300215,2208230009,d_d000621,27,앤유 피시방,2022-08-23 00:00:00,2022-08-23 00:00:00,NaN,제주특별자치도 제주시 연동,126.496886,...,1,2.0,7.0,1,제주특별자치도 제주시 연동,2,1,3,210.0,4
34718,1300216,2208230007,d_d000621,25,전주아줌마두루치기,2022-08-23 00:00:00,2022-08-23 00:00:00,제주특별자치도 제주시 신광로4길 25,제주특별자치도 제주시 연동 251-59,126.487416,...,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 25,0,1,11,30.0,9


In [ ]:
sangum = df[df['VISIT_AREA_NM'] == '한림리조트']
sangum[['VISIT_CHC_REASON_CD', 'VISIT_MAIN_REASON']]

,VISIT_CHC_REASON_CD,VISIT_MAIN_REASON
116,8,11
117,8,11
118,8,11
119,2,11
120,2,11
121,2,11
122,11,11
123,11,11
124,11,11
125,11,11


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34720 entries, 0 to 34719
Data columns (total 74 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Unnamed: 0                 34720 non-null  int64  
 1   VISIT_AREA_ID              34720 non-null  int64  
 2   TRAVEL_ID                  34720 non-null  object 
 3   VISIT_ORDER                34720 non-null  int64  
 4   VISIT_AREA_NM              34720 non-null  object 
 5   VISIT_START_YMD            34720 non-null  object 
 6   VISIT_END_YMD              34720 non-null  object 
 7   ROAD_NM_ADDR               29536 non-null  object 
 8   LOTNO_ADDR                 34250 non-null  object 
 9   X_COORD                    34720 non-null  float64
 10  Y_COORD                    34720 non-null  float64
 11  ROAD_NM_CD                 0 non-null      float64
 12  LOTNO_CD                   0 non-null      float64
 13  POI_ID                     13740 non-null  obj

# 2. 데이터 저장

In [ ]:
df.to_csv('Data/Final/final_04.csv', index = False)

In [ ]:
data = pd.read_csv('Data/Final/final_04.csv')

In [ ]:
data

,Unnamed: 0,VISIT_AREA_ID,TRAVEL_ID,VISIT_ORDER,VISIT_AREA_NM,VISIT_START_YMD,VISIT_END_YMD,ROAD_NM_ADDR,LOTNO_ADDR,X_COORD,...,TRAVEL_MOTIVE_1,TRAVEL_MOTIVE_2,TRAVEL_MOTIVE_3,TRAVEL_COMPANIONS_NUM,ADDR,RES_CAFE,VISIT_COUNT,VISIT_AREA_MAIN_TYPE,AVG_TIME_MIN,VISIT_MAIN_REASON
0,0,2208190004,d_d000296,14,산굼부리,2022-08-19 00:00:00,2022-08-19 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,3,7.0,1.0,3,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1,60.0,1
1,3658,2210170005,d_d003259,4,산굼부리,2022-10-17 00:00:00,2022-10-17 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,2,1.0,3.0,1,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1,60.0,1
2,9721,2209280002,d_d001522,21,산굼부리,2022-09-28 00:00:00,2022-09-28 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,2,3.0,1.0,5,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1,60.0,1
3,28034,2211010002,d_d010245,22,산굼부리,2022-11-01 00:00:00,2022-11-01 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,1,2.0,6.0,1,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1,60.0,1
4,51305,2210240001,d_d003054,8,산굼부리,2022-10-24 00:00:00,2022-10-24 00:00:00,NaN,제주특별자치도 제주시 조천읍 교래리 166-2,126.693832,...,3,2.0,5.0,1,제주특별자치도 제주시 조천읍 교래리 166-2,2,99,1,60.0,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
34715,1300145,2208210007,d_d000621,5,미니K코인노래연습장,2022-08-21 00:00:00,2022-08-21 00:00:00,제주특별자치도 제주시 국기로 5,제주특별자치도 제주시 연동 1457,126.490036,...,1,2.0,7.0,1,제주특별자치도 제주시 국기로 5,2,1,4,30.0,9
34716,1300205,2208210008,d_d000621,6,장동민의크라우드pc방,2022-08-21 00:00:00,2022-08-21 00:00:00,제주특별자치도 제주시 건주로4길 10-4,제주특별자치도 제주시 도련일동 1939-3,126.579895,...,1,2.0,7.0,1,제주특별자치도 제주시 건주로4길 10-4,2,1,3,150.0,9
34717,1300215,2208230009,d_d000621,27,앤유 피시방,2022-08-23 00:00:00,2022-08-23 00:00:00,NaN,제주특별자치도 제주시 연동,126.496886,...,1,2.0,7.0,1,제주특별자치도 제주시 연동,2,1,3,210.0,4
34718,1300216,2208230007,d_d000621,25,전주아줌마두루치기,2022-08-23 00:00:00,2022-08-23 00:00:00,제주특별자치도 제주시 신광로4길 25,제주특별자치도 제주시 연동 251-59,126.487416,...,1,2.0,7.0,1,제주특별자치도 제주시 신광로4길 25,0,1,11,30.0,9


In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34720 entries, 0 to 34719
Data columns (total 74 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Unnamed: 0                 34720 non-null  int64  
 1   VISIT_AREA_ID              34720 non-null  int64  
 2   TRAVEL_ID                  34720 non-null  object 
 3   VISIT_ORDER                34720 non-null  int64  
 4   VISIT_AREA_NM              34720 non-null  object 
 5   VISIT_START_YMD            34720 non-null  object 
 6   VISIT_END_YMD              34720 non-null  object 
 7   ROAD_NM_ADDR               29536 non-null  object 
 8   LOTNO_ADDR                 34250 non-null  object 
 9   X_COORD                    34720 non-null  float64
 10  Y_COORD                    34720 non-null  float64
 11  ROAD_NM_CD                 0 non-null      float64
 12  LOTNO_CD                   0 non-null      float64
 13  POI_ID                     13740 non-null  obj